beacon-chain/state/v1/getters_block_test.go

```diff
@@ -0,0 +1,60 @@
+package v1
+
+import (
+	"testing"
+
+	pbp2p "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	v1alpha1 "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/testutil/require"
+)
+
+func TestBeaconState_LatestBlockHeader(t *testing.T) {
+	s, err := InitializeFromProto(&pbp2p.BeaconState{})
+	require.NoError(t, err)
+	got := s.LatestBlockHeader()
+	require.DeepEqual(t, (*v1alpha1.BeaconBlockHeader)(nil), got)
+
+	want := &v1alpha1.BeaconBlockHeader{Slot: 100}
+	s, err = InitializeFromProto(&pbp2p.BeaconState{LatestBlockHeader: want})
+	require.NoError(t, err)
+	got = s.LatestBlockHeader()
+	require.DeepEqual(t, want, got)
+
+	// Test copy does not mutate.
+	got = &v1alpha1.BeaconBlockHeader{Slot: 101}
+	require.DeepNotEqual(t, want, got)
+}
+
+func TestBeaconState_BlockRoots(t *testing.T) {
+	s, err := InitializeFromProto(&pbp2p.BeaconState{})
+	require.NoError(t, err)
+	got := s.BlockRoots()
+	require.DeepEqual(t, ([][]byte)(nil), got)
+
+	want := [][]byte{{'a'}}
+	s, err = InitializeFromProto(&pbp2p.BeaconState{BlockRoots: want})
+	require.NoError(t, err)
+	got = s.BlockRoots()
+	require.DeepEqual(t, want, got)
+
+	// Test copy does not mutate.
+	got = [][]byte{{'b'}}
+	require.DeepNotEqual(t, want, got)
```

Comment:
So the mutation we want to test is that we want to be sure that the pointer references in an object don't still point to the same thing. Especially with slices and their underlying array.  In this example you have here, you are reassigning got entirely, which would not test that the pointers do not overlap.

Maybe we can try something like 
```suggestion
	// Test copy does not mutate.
	got[0][0] = 'b'
	require.DeepNotEqual(t, want, got)
```

One thing on our unwritten TODO list is to have some "smart" asserts that copied objects share zero pointers. That is the ultimate check that a copied object would not mutate the original object or any of its references.

Reply:
Edit: If `got[0]` and `want[0]` shared the same slice/pointer, mutating `got[0][0]` would have also mutated `want[0][0]`

Reply:
Good point, I completely missed this